### PR TITLE
Adblocking

### PR DESCRIPTION
**Pull request by leggsimon**
/cc @adgad @commuterjoy 

This adds an adblocking detection tool similar to Sourcepoint that we will run in parallel to compare numbers.

### CODE REVIEW

File: src/analytics/adblocking.js [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+										context: {
+												provider: 'blockAdBlock'
+										}
+								},
+								bubbles: true
+						}));
+				}
+		};
+
+		window.BlockAdBlock = BlockAdBlock;
```
Comment (adgad) [inline src/analytics/adblocking.js]: Do we need to set global variables here?

(The answer should be no)

File: src/analytics/adblocking.js [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+				};
+				if (options !== undefined) {
+						this.setOption(options);
+				}
+				var self = this;
+				var eventCallback = function() {
+						setTimeout(function() {
+								if (self._options.checkOnLoad === true) {
+										if (self._options.debug === true) {
+												self._log('onload->eventCallback', 'A check loading is launched');
```
Comment (adgad) [inline src/analytics/adblocking.js]: Could maybe get rid of these log statements too.

**Comment by leggsimon** [comment]
Updated to remove global variables and in general just make it all that little bit nicer 😊 

@adgad @commuterjoy

**Comment by adgad** [comment]
LGTM

**Pull request merged**